tests/core/policies/test_rule_policy.py

```diff
@@ -656,6 +656,48 @@ def test_all_policy_attributes_are_persisted(tmpdir: Path):
     assert persisted_policy._enable_fallback_prediction == enable_fallback_prediction
 
 
+async def test_rule_policy_finetune(
```

Comment:
could you please add a test that contradiction is raised if you add contradicting rule in fine-tune

Reply:
Added